WordPress/Sniffs/WP/DeprecatedFunctionsSniff.php

```diff
@@ -0,0 +1,1097 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of various deprecated WordPress functions and suggests alternatives.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.11.0
+ */
+class WordPress_Sniffs_WP_DeprecatedFunctionsSniff extends WordPress_AbstractFunctionRestrictionsSniff {
+
+	/**
+	 * Mimimum WordPress version.
+	 *
+	 * Example: '4.3'
+	 *
+	 * @var string WordPress versions.
+	 */
+	public $minimum_supported_version = 4.3;
+
+	/**
+	 * List of deprecated functions with alternative when available.
+	 *
+	 * To be updated after every major release.
```

Comment:
I would suggest to add a note which documents when the list was last updated and on which version / commit the last update was based, so the next person updating it knows where to start from.